### PR TITLE
Add support for CMake

### PR DESCRIPTION
**Pull request by dbrgn**
Building and installing:

```
$ mkdir build && cd build && cmake .. -DCMAKE_INSTALL_PREFIX=$(pwd)/usr
-- The CXX compiler identification is GNU 10.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build

$ make
Scanning dependencies of target fontobene_qt5
[  0%] Built target fontobene_qt5
Scanning dependencies of target fontobene_qt5_tests
[ 50%] Building CXX object tests/CMakeFiles/fontobene_qt5_tests.dir/main.cpp.o
[100%] Linking CXX executable fontobene-qt5-tests
[100%] Built target fontobene_qt5_tests

$ make install
[  0%] Built target fontobene_qt5
[100%] Built target fontobene_qt5_tests
Install the project...
-- Install configuration: ""
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyphlist.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/font.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyph.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyphlistaccessor.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/exception.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/helpers.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/header.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/fontobene.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyphlistcache.h
```

The resulting files:

```
$ tree usr
usr
└── include
    └── fontobene-qt5
        ├── exception.h
        ├── font.h
        ├── fontobene.h
        ├── glyph.h
        ├── glyphlist.h
        ├── glyphlistaccessor.h
        ├── glyphlistcache.h
        ├── header.h
        └── helpers.h

2 directories, 9 files
```

### CODE REVIEW

**Comment by dbrgn** [comment]
Building and installing:

```
$ mkdir build && cd build && cmake .. -DCMAKE_INSTALL_PREFIX=$(pwd)/usr
-- The CXX compiler identification is GNU 10.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build
$ make
Scanning dependencies of target fontobene_qt5
[  0%] Built target fontobene_qt5
Scanning dependencies of target fontobene_qt5_tests
[ 50%] Building CXX object tests/CMakeFiles/fontobene_qt5_tests.dir/main.cpp.o
[100%] Linking CXX executable fontobene-qt5-tests
[100%] Built target fontobene_qt5_tests
$ $ make install
[  0%] Built target fontobene_qt5
[100%] Built target fontobene_qt5_tests
Install the project...
-- Install configuration: ""
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyphlist.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/font.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyph.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyphlistaccessor.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/exception.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/helpers.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/header.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/fontobene.h
-- Installing: /home/danilo/Projects/librepcb/LibrePCB/libs/fontobene-qt5/build/usr/include/fontobene-qt5/glyphlistcache.h
```

The resulting files:

```
$ tree usr
usr
└── include
    └── fontobene-qt5
        ├── exception.h
        ├── font.h
        ├── fontobene.h
        ├── glyph.h
        ├── glyphlist.h
        ├── glyphlistaccessor.h
        ├── glyphlistcache.h
        ├── header.h
        └── helpers.h

2 directories, 9 files
```

**Review by ubruhin — DISMISSED** [review]
Nice! :+1:

**Comment by ubruhin** [comment]
The CI job names are too long to be fully displayed in the UI. And actually I'd focus more on the used Ubuntu version than on the cmake version (older Ubuntu version also tests with an older Qt version, which is useful as well). Maybe something like that?

- Ubuntu 20.04 qmake
- Ubuntu 20.04 cmake
- Ubuntu 16.04 cmake

**Comment by dbrgn** [comment]
Yeah, I noticed that as well. Plus, some tests fail.

I'll take a look tomorrow.

**Comment by ubruhin** [comment]
> Yeah, I noticed that as well. Plus, some tests fail. I'll take a look tomorrow.

You might want to take a look at https://github.com/LibrePCB/parseagle/pull/4 :grin:

**Comment by dbrgn** [comment]
@ubruhin updated and fixed! (Support for C++11 needed to be enforced in order to work on Ubuntu 16.04.)

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# CMake version configuration
+cmake_minimum_required(VERSION 3.5...3.19)
+if(${CMAKE_VERSION} VERSION_LESS 3.12)
+    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+endif()
+
+# Define project
+if(${CMAKE_VERSION} VERSION_LESS 3.12)
```
Comment (ubruhin) [inline CMakeLists.txt]: Why this if/else, can't we simply always omit `DESCRIPTION` and `HOMEPAGE_URL`?

Comment (dbrgn) [reply CMakeLists.txt]: It could be useful for newer CMake versions, but we could also remove it if you prefer that.

Comment (ubruhin) [reply CMakeLists.txt]: Okay so if it has no clear advantage, honestly I'd prefer to remove it to keep the cmake config simple (avoiding redundancy) :wink:

Comment (dbrgn) [reply CMakeLists.txt]: Done!

**Review by ubruhin — APPROVED** [review]
:tada:

**Pull request merged**